api/server/router/build/build_routes.go

```diff
@@ -61,9 +61,21 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 		BuildID:        r.FormValue("buildid"),
 	}
 
-	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
-		// SecurityOpt only supports "credentials-spec" on Windows, and not used on other platforms.
-		return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+	if len(options.SecurityOpt) > 0 {
+		// SecurityOpt only supports "credentialspec" on Windows, and not used on other platforms.
+		if runtime.GOOS != "windows" {
+			return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+		}
+		for _, secOpt := range options.SecurityOpt {
+			opt, _, ok := strings.Cut(secOpt, "=")
+			if !ok {
+				return nil, invalidParam{errors.New("invalid security option: no equals sign in supplied value " + secOpt)}
+			}
+			// FIXME(thaJeztah): options should not be case-insensitive
+			if !strings.EqualFold(opt, "credentialspec") {
+				return nil, invalidParam{errors.New("security option not supported: " + opt)}
+			}
+		}
 	}
```

Comment:
Validating that security options are syntactically well-formed (contains an equal sign, etc.) is one thing, but I don't think validating which options are _semantically_ acceptable is within the scope of an API router's responsibility. The most SOLID architecture would be to push that responsibility into each builder implementation so each implementation _could_ apply different validation rules, though that would be very repetitive today when the rules are the same for all builders on a given platform. How about the more pragmatic approach of defining `type SecurityOpts []string` with a `Validate() error` method, following the lead of `api/types/container.Isolation`? Extracting the validation logic to a dedicated function would make it more testable with an isolated unit test; even more so if the function was referentially transparent: `Validate(goos string) error`